Modules/JavaScript/Classes-Constructor-New/homework.js

```diff
@@ -0,0 +1,84 @@
+console.log(`
+  Task.
+
+    Create a Vector object that supports:
+     1. add (same length)
+     2. subtract
+     3. dot
+     4. norm
+     5. If you try to add, subtract, or dot two vectors with different lengths, you must throw an error.
+     6. toString
+     7. equals
+`);
+
+class Vector {
+  list = [];
+
+  constructor(list) {
+    this.list = list;
+  }
+
+  add(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val + vector.list[index]));
+  }
+  
+  subtract(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val - vector.list[index]));
+  }
+  
+  dot(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return this.list.reduce((sum, val, index) => {
+      console.log('  ----- val=', val, 'vector.list[index]=', vector.list[index], 'addThis=', (val * vector.list[index]))
+      return sum + (val * vector.list[index]);
+    }, 0)
+  }
+  
+  norm() {
+    const sum = this.list.reduce((a, b) => {
+      console.log('  ----- val^2=', Math.pow(b, 2));
+      return a + (b * 2);
+    }, 0);
+    return Math.sqrt(sum);
+  }
+
+  toString() {
+    return `(${this.list.toString()})`;
+  }
+
+  equals(vector) {
+    if (this.list.length === vector.list.length) {
```

Comment:
Modify the validator and use it here. In case you would need to change the validation logic in the future, you wouldn't have to update this part separately